### PR TITLE
Upgrade guide for the obsoleted configure class

### LINKED ISSUE #1282: No Doco for this obsolete message 'Configure' is obsolete: 'This is no longer a public API'
'Configure' is obsolete: 'This is no longer a public API. Will be removed in version 7.0.0.'

I looked through the [upgrade guide ](http://docs.particular.net/nservicebus/upgrades/5to6 ) and wasn't able to find examples of what to do.

I looked at the NServiceBus v6 code no hints there

``` C#
    [ObsoleteEx(
        Message = "This is no longer a public API",
        RemoveInVersion = "7.0",
        TreatAsErrorFromVersion = "6.0")]
    public class Configure
    {
    }
```

While I understand that it is likely that this was only used internally It would be nice to have some kind of clue as to what to replace it's usage with in downstream assemblies.

/cc @Particular/nservicebus-maintainers

### PR DESCRIPTION
**Pull request by andreasohlund**
Fixes #1282

### CODE REVIEW

File: nservicebus/upgrades/5to6.md [hunk 1]
```diff
@@ -680,3 +680,22 @@ snippet: 5to6-ExplicitProperties
 To access it at runtime create a new [`Feature`](/nservicebus/pipeline/features.md) and put the following code in the `.Setup` method 
 
 snippet: 5to6-IConfigureComponentsNotInjected
+
+# Configure type no longer available
+
+In Version 5 the configure type was used to provide runtime access to the local endpoint address, scanned types etc via dependency injection.
```
Comment (SimonCropp) [inline nservicebus/upgrades/5to6.md]: perhaps end with some preamble? "these things can now be accesses with the following"?

File: nservicebus/upgrades/5to6.md [hunk 2]
```diff
@@ -680,3 +680,22 @@ snippet: 5to6-ExplicitProperties
...
+## Settings
+
+Settings can be accessed via the `FeatureConfigurationContext`, see [features](/nservicebus/pipeline/features.md) for more details. Runtime access via the container is provided by taking a dependency on the `ReadOnlySettings` type.
+
+## Builder
+
+This is no longer supported. It is advised to, instead of using `IBuilder` directly, use dependency injection via the container of choice.
+## Scanned types
+
+Access to types found during assembly scanning is provided via `Settings.GetAvailableTypes()`.
```
Comment (SimonCropp) [inline nservicebus/upgrades/5to6.md]: link to more doco

File: nservicebus/upgrades/5to6.md [hunk 3]
```diff
@@ -680,3 +680,22 @@ snippet: 5to6-ExplicitProperties
...
+
+## Local address
+
+Access to the endpoint address is provided via `Settings.LocalAddress()`.
```
Comment (SimonCropp) [inline nservicebus/upgrades/5to6.md]: link to more doco

File: nservicebus/upgrades/5to6.md [hunk 4]
```diff
@@ -680,3 +680,22 @@ snippet: 5to6-ExplicitProperties
...
+
+In Version 5 the configure type was used to provide runtime access to the local endpoint address, scanned types etc via dependency injection.
+
+## Settings
+
+Settings can be accessed via the `FeatureConfigurationContext`, see [features](/nservicebus/pipeline/features.md) for more details. Runtime access via the container is provided by taking a dependency on the `ReadOnlySettings` type.
+
+## Builder
+
+This is no longer supported. It is advised to, instead of using `IBuilder` directly, use dependency injection via the container of choice.
```
Comment (SimonCropp) [inline nservicebus/upgrades/5to6.md]: link to more doco

File: nservicebus/upgrades/5to6.md (hunk 1 above)
Comment (SimonCropp) [inline nservicebus/upgrades/5to6.md]: > the `Configure` type

**Comment by SimonCropp** [comment]
@andreasohlund can you please finish this

File: nservicebus/upgrades/5to6.md (hunk 1 above)
Comment (andreasohlund) [inline nservicebus/upgrades/5to6.md]: Fixed

File: nservicebus/upgrades/5to6.md (hunk 2 above)
Comment (andreasohlund) [inline nservicebus/upgrades/5to6.md]: fixed

**Comment by andreasohlund** [comment]
@SimonCropp updated, wip removed, should be good to go

**Comment by SimonCropp** [comment]
@andreasohlund awesome. thanks

**Pull request merged**